runtime/buffer.h

```diff
@@ -299,7 +300,7 @@ class raw_buffer {
 
     dim(d).set_bounds(min, max);
     // Crops can't span a folding boundary if they move the base pointer.
-    assert(base == nullptr || offset == 0 || !dim(d).is_folded());
+    assert(base == nullptr || offset == 0 || !dim(d).is_folded(min - old_min, max - old_min));
```

Comment:
I probably keep dropping it because it somehow doesn't look right to me, it looks like it's passing "vectors" to arguments that should be "positions". 

Also, I'm going to guess this is going to break builds when the compiler checks for unused variables and asserts are disabled.

Now that we have `is_folded(min, max)` can we write the assert in a more clear way by doing it before we call `set_bounds`?